sklearn/neighbors/_base.py

```diff
@@ -385,6 +387,17 @@ def _radius_neighbors_from_graph(graph, radius, return_distance):
 
 class NeighborsBase(MultiOutputMixin, BaseEstimator, metaclass=ABCMeta):
     """Base class for nearest neighbors estimators."""
+    _parameter_constraints = {
+        "n_neighbors": [Interval(Integral, 1, None, closed="left")],
+        "radius": [Interval(Integral, 1, None, closed="left"), None],
+        "weights": [StrOptions({"uniform", "distance"}), callable],
+        "algorithm": [StrOptions({"auto", "ball_tree", "kd_tree", "brute"}), callable],
+        "leaf_size": [Interval(Integral, 1, None, closed="left")],
+        "p": [Interval(Integral, 1, None, closed="left")],
+        "metric": [StrOptions({"minkowski"}), callable],
```

Comment:
There are many string options for `metric`. They are all listed in VALID_METRICS.
```suggestion
        "metric": [StrOptions({metric for group in VALID_METRICS.values() for metric in group}), callable],
```